x-pack/plugins/ingest_pipelines/__jest__/client_integration/ingest_pipelines_clone.test.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+import { act } from 'react-dom/test-utils';
+
+import { setupEnvironment, pageHelpers } from './helpers';
+import { PipelineFormTestBed } from './helpers/pipeline_form.helpers';
+import { PIPELINE_TO_CLONE } from './helpers/pipelines_clone.helpers';
+
+const { setup } = pageHelpers.pipelinesClone;
+
+jest.mock('@elastic/eui', () => ({
+  ...jest.requireActual('@elastic/eui'),
+  // Mocking EuiCodeEditor, which uses React Ace under the hood
+  EuiCodeEditor: (props: any) => (
+    <input
+      data-test-subj={props['data-test-subj']}
+      onChange={(syntheticEvent: any) => {
+        props.onChange(syntheticEvent.jsonString);
+      }}
+    />
+  ),
+}));
+
+describe('<PipelinesClone />', () => {
+  let testBed: PipelineFormTestBed;
+
+  const { server, httpRequestsMockHelpers } = setupEnvironment();
+
+  afterAll(() => {
+    server.restore();
+  });
+
+  beforeEach(async () => {
+    httpRequestsMockHelpers.setLoadPipelineResponse(PIPELINE_TO_CLONE);
+
+    await act(async () => {
+      testBed = await setup();
+      await testBed.waitFor('pipelineForm');
```

Comment:
Just a note about `waitFor`. I realized it might not have been the best idea to add `setTimout` calls inside our tests (which is what `waitFor` does).
It _seems_ (but I am still investigating) that some of the timing issues we are getting in our tests come from `setTimeout` calls inside the code being tested (and the form lib has some).
Once I get to the bottom of it I'll share my findings 😊 For now let's keep with this small hack.